spec/controllers/pieces_controller_spec.rb

```diff
@@ -147,13 +147,31 @@
 
       sign_in @player2
 
-      put :update, params: {id: black_queen.id, x_position: 1, y_position: 6, format: :js }
+      put :update, params: { id: black_queen.id, x_position: 1, y_position: 6, format: :js }
       black_queen.reload
       @game.reload
       expect(black_queen.x_position).to eq 1
       expect(black_queen.y_position).to eq 6
       expect(@game.winner).to eq @player2
     end
+
+    it 'puts game in check when black piece takes a white piece and puts white king in check' do
+      white_king = create(:king, x_position: 0, y_position: 4, game_id: @game.id, piece_number: 4)
+      white_pawn = create(:pawn, x_position: 0, y_position: 5, game_id: @game.id, piece_number: 5)
+      black_rook = create(:rook, x_position: 0, y_position: 7, game_id: @game.id, piece_number: 6)
+      create(:move, game_id: @game.id, piece_id: white_pawn.id, start_piece: 5)
+
+      sign_in @player2
+
+      black_rook.move_to!(0, 5)
+      white_pawn.reload
+      black_rook.reload
+      @game.reload
+
+      expect(black_rook.x_position).to eq 0
+      expect(black_rook.y_position).to eq 5
+      expect(@game.check?(!black_rook.is_white?)).to eq true
+    end
```

Comment:
I still think that this is checking the` check?` method in the game.rb file not the `check_test` method in the update controller since `move_to!` just forces the piece to move it does not check any of the other methods inside the update controller. I will download and work on this test also.

Reply:
Yes, but `move_to!` and `game.check?` are both called from piece#update. Since this test is updating a piece, I think it fits here. If you come up with something better, by all means push it up.

Reply:
Yup working on it now. And yes the are both called from within piece#update but I do not think that it means it is checking the other methods in piece#update.